domain/application/service/unit.go

```diff
@@ -26,24 +26,38 @@ import (
 // UnitState describes retrieval and persistence methods for
 // units.
 type UnitState interface {
-	// AddUnits adds the specified units to the application.
-	AddUnits(context.Context, string, coreapplication.ID, []application.AddUnitArg) error
-
-	// InsertCAASUnit inserts the specified CAAS application unit, returning an
-	// error satisfying [applicationerrors.UnitAlreadyExists] if the unit
-	// exists.
-	InsertCAASUnit(context.Context, coreapplication.ID, application.RegisterCAASUnitArg) error
+	// AddIAASUnits adds the specified units to the application.
+	// If the application is not found, an error satisfying [applicationerrors.ApplicationNotFound] is returned.
+	// If any of the units already exists, an error satisfying [applicationerrors.UnitAlreadyExists] is returned.
+	AddIAASUnits(context.Context, string, coreapplication.ID, ...application.AddUnitArg) error
+
+	// AddCAASUnits adds the specified units to the application.
+	// If the application is not found, an error satisfying [applicationerrors.ApplicationNotFound] is returned.
+	// If any of the units already exists, an error satisfying [applicationerrors.UnitAlreadyExists] is returned.
+	AddCAASUnits(context.Context, string, coreapplication.ID, ...application.AddUnitArg) error
+
+	// InsertMigratingIAASUnits inserts the fully formed units for the specified IAAS application.
+	// This is only used when inserting units during model migration.
+	// If the application is not found, an error satisfying [applicationerrors.ApplicationNotFound] is returned.
+	// If any of the units already exists, an error satisfying [applicationerrors.UnitAlreadyExists] is returned.
+	InsertMigratingIAASUnits(context.Context, coreapplication.ID, ...application.InsertUnitArg) error
+
+	// InsertMigratingCAASUnits inserts the fully formed units for the specified CAAS application.
+	// This is only used when inserting units during model migration.
+	// If the application is not found, an error satisfying [applicationerrors.ApplicationNotFound] is returned.
+	// If any of the units already exists, an error satisfying [applicationerrors.UnitAlreadyExists] is returned.
+	InsertMigratingCAASUnits(context.Context, coreapplication.ID, ...application.InsertUnitArg) error
```

Comment:
I am not a fan of these being public methods, but we can change these in subsequent PRs.

Reply:
These will go away with the next refactoring PR.